### PR TITLE
Create pull request template

### PR DESCRIPTION
**Pull request by xpaczka**
## Pull request template

This PR creates a pull request template that will be used for all pull requests, to make them consistent and always provide enough information about what was changes / what was added

**The PR template looks like this:**

## Issues connected

Resolves

## What has been done

- Change 1

## Testing

- [ ] Test 1

## Screenshots / images / videos

Please provide assets necessary for the PR (remove if not applicable)

**Open to suggestions, what should be changed / what should be added**

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+Please provide a list of issues connected to this PR (resolves / closes or ref if not closing the selected issue)
+
+## Description
+
+Please include a summary of the changes
+
+## Pull request type
+
+- [ ] Bug fix
+- [ ] New feature
```
Comment (michalinacienciala) [inline .github/pull_request_template.md:12]: Maybe we could add another item for dev-ops related PRs (like the one we're looking at currently)? Or would those fall under `New feature`/`Bug fix` or `Other`?

File: .github/workflows/create-pr-template.yml
Comment (michalinacienciala) [inline .github/workflows/create-pr-template.yml:1]: We don't need this workflow to automatically add the template as a suggested PR description, it should work just with the `.github/pull_request_template.md`.

**Review by michalinacienciala — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md (hunk 1 above)
Comment (xpaczka) [reply .github/pull_request_template.md:12]: I'm really open to all suggetions - these are the things that came to my mind, but will be happy to add more, especially to make `Other` less reliable

**Review by jagodarybacka — COMMENTED** [review]
So what I would like to achieve with a PR template is both:
- having unified structure of PR descriptions
- improving or keeping the same the effort of creating the PR description while providing the value to PR reviewers

I would say that to achieve the points above we should aim to have to type as little text as possible while including as much context as possible.

I feel like this template contains too much text and includes details that are obfuscating the message of the PR.

My suggestions:
- type the word "Resolves " instead of describing what this section is about - will save us time removing that text. In general I think we don't need to have the descriptions of these sections - either we will be removing them (additional time to remove that text) or we will leave them in the cretaed PR (additional effort for the reader to ignore that text while reading)
- we just need one bullet point or checklist item prepared in the template to be able to add next by clicking "enter" while typing
- "description" and "what has been done" are basically the same thing most of the time
- why do we need to check the type of the PR? What kind of value is it providing? I believe it should be obvious looking at the title or description. I would find it valuable if we could filter by the type (so it would be a label for example) but idk if this is necessary
- screenshots are optional and IMO they should be below the testing checklist (but its just my preference to keep the text content together and see the testing progress first)

In general I like the template idea but we should be careful while designing that thing to avoid adding more work than we had before 😄

**Review by jagodarybacka — APPROVED** [review]
I like it now, do we want to undraft?

**Pull request merged**